app/src/main/java/org/sopt/and/presentation/viewmodel/MyviewViewModel.kt

```diff
@@ -0,0 +1,36 @@
+package org.sopt.and.presentation.viewmodel
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import org.sopt.and.data.repository.MyviewRepository
+import org.sopt.and.data.repository.SharedPreferencesHelper
+import javax.inject.Inject
+
+@HiltViewModel
+class MyviewViewModel @Inject constructor(
+    private val sharedPreferencesHelper: SharedPreferencesHelper,
+    private val myviewRepository: MyviewRepository
+) : ViewModel() {
+    private val _hobbyResult = MutableLiveData<Result<String>>()
+    val hobbyResult: LiveData<Result<String>> get() = _hobbyResult
+
+    fun getHobby() {
+        viewModelScope.launch {
+            val token = getTokenForRequest()
+            if (token != null) {
+                val result = myviewRepository.getHobby(token)
+                _hobbyResult.postValue(result)
```

Comment:
의심되는 부분은 여기인데 아래처럼 바꿔도 안되면 ui 쪽에 문제인거 같네요
```suggestion
                   val result = withContext(Dispatchers.IO) { myviewRepository.getHobby(token) } 
            _hobbyResult.postValue(result)

```

Reply:
LiveData 는 스레드의 자원을 많이 잡아먹어서 stateFlow 것을 한번 사용해 보시는것도 추천드려요 
https://readystory.tistory.com/207

Reply:
라이브 데이터에 관한 문제로 생길 수 도 있다고 하니 아래 한번 참고해보셔도 좋을거같아요 

https://jminie.tistory.com/188#3.2.%20%EB%B9%84%EB%8F%99%EA%B8%B0%20%EC%9E%91%EC%97%85%20%EC%B8%A1%EB%A9%B4%EC%97%90%EC%84%9C%20LiveData%EC%9D%98%20%EB%8B%A8%EC%A0%90

Reply:
[3]
LiveData 대신 StateFlow를 사용하는 방법도 있을 것 같습니다 ㅎㅎ 두 개는 어떤 차이가 있을까요?